content/en/blog/_posts/2023-04-05-oci-security-profiles.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: blog
+title: "Using OCI artifacts to distribute security profiles for seccomp, SELinux and AppArmor"
+date: 2023-04-05
+slug: oci-security-profiles
+---
+
+**Author**: Sascha Grunert
+
+The [Security Profiles Operator (SPO)][spo] makes managing seccomp, SELinux and
+AppArmor profiles within Kubernetes easier than ever. It allows cluster
+administrators to define the profiles in a predefined custom resource YAML,
+which then gets distributed by the SPO into the whole cluster. Modification and
+removal of the security profiles are managed by the operator in the same way,
+but that’s a small subset of its capabilities.
+
+[spo]: https://github.com/kubernetes-sigs/security-profiles-operator
+
+Another core feature of the SPO is being able to stack seccomp profiles. This
+means that users can define a `baseProfileName` in the YAML specification, which
+then gets automatically resolved by the operator and combines the syscall rules.
+If a base profile has another `baseProfileName`, then the operator will
+recursively resolve the profiles up to a certain depth. A common use case is to
+define base profiles for low level container runtimes (like [runc][runc] or
+[crun][crun]) which then contain syscalls which are required in any case to run
+the container. Alternatively, application developers can define seccomp base
+profiles for their standard distribution containers and stack dedicated profiles
+for the application logic on top. This way developers can focus on maintaining
+seccomp profiles which are way simpler and scoped to the application logic,
+without having a need to take the whole infrastructure setup into account.
+
+[runc]: https://github.com/opencontainers/runc
+[crun]: https://github.com/containers/crun
+
+But how to maintain those base profiles? For example, the amount of required
+syscalls for a runtime can change over its release cycle in the same way it can
+change for the main application. Base profiles have to be available in the same
+cluster, otherwise the main seccomp profile will fail to deploy. This means that
+they’re tightly coupled to the main application profiles, which acts against the
+main idea of base profiles. Distributing and managing them as plain files feels
+like an additional burden we have to solve.
+
+## OCI artifacts to the rescue
+
+The [latest release][spo-latest] of the Security Profiles Operator supports
+managing base profiles as OCI artifacts! Imagine OCI artifacts as lightweight
+container images, storing files in layers in the same way images do, but without
+a process to be executed. Those artifacts can be used to store security profiles
+like regular container images in compatible registries. This means they can be
+versioned, namespaced and annotated similar to regular container images.
+
+[spo-latest]: https://github.com/kubernetes-sigs/security-profiles-operator/releases/latest
+
+To see how that works in action, we have to specify a `baseProfileName` prefixed
```

Comment:
```suggestion
To see how that works in action,  specify a `baseProfileName` prefixed
```
We [avoid using we](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we) in tech docs and in other contexts where it's not clear who the “we” refers to.